diff --git a/Makefile b/Makefile
index 22ff3f5..183c0bb 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,16 @@
-hemeXtract:
-	g++ -Wall -O3 hemeXtract.cc -o hemeXtract
-debug:
-	g++ -Wall -g hemeXtract.cc -o hemeXtract
+
+CXX=g++ -Wall
+CXXFLAGS=-O3
+CXXFLAGS_DEBUG=-g
+
+SRC=hemeXtract.cc
+
+hemeXtract: $(SRC)
+	$(CXX) $(CXXFLAGS) -o $@ $< -ltirpc
+
+debug: $(SRC)
+	$(CXX) $(CXXFLAGS_DEBUG) -o $@ $< -ltirpc
+
+.phony: clean
+clean:
+	rm -f hemeXtract
\ No newline at end of file
